src/app/server/EnhancedSetupFlowProvider.h

```diff
@@ -0,0 +1,108 @@
+     * @return CHIP_ERROR On success returns CHIP_NO_ERROR, otherwise returns an error code.
+     */
+    virtual CHIP_ERROR HasTermsAndConditionsRequiredAcknowledgementsBeenAccepted(bool & outAccepted) const = 0;
+
+    /**
+     * @brief Checks if the required terms and conditions acknowledgements version has been accepted.
+     *
+     * @param[out] outAccepted true if the required acknowledgements version has been accepted, false otherwise.
+     * @return CHIP_ERROR On success returns CHIP_NO_ERROR, otherwise returns an error code.
+     */
+    virtual CHIP_ERROR HasTermsAndConditionsRequiredAcknowledgementsVersionBeenAccepted(bool & outAccepted) const = 0;
+
+    /**
+     * @brief Retrieves the required terms and conditions acknowledgements.
+     *
+     * @param[out] outValue The version of the required acknowledgements.
+     * @return CHIP_ERROR On success returns CHIP_NO_ERROR, otherwise returns an error code.
+     */
+    virtual CHIP_ERROR GetTermsAndConditionsRequiredAcknowledgements(uint16_t & outValue) const = 0;
+
+    /**
+     * @brief Retrieves the required terms and conditions acknowledgements version.
+     *
+     * @param[out] outValue The outValue of the required acknowledgements version.
+     * @return CHIP_ERROR On success returns CHIP_NO_ERROR, otherwise returns an error code.
+     */
+    virtual CHIP_ERROR GetTermsAndConditionsRequiredAcknowledgementsVersion(uint16_t & outValue) const = 0;
+
+    /**
+     * @brief Retrieves the accepted terms and conditions acknowledgements.
+     *
+     * @param[out] outValue The outValue of the accepted acknowledgements.
+     * @return CHIP_ERROR On success returns CHIP_NO_ERROR, otherwise returns an error code.
+     */
+    virtual CHIP_ERROR GetTermsAndConditionsAcceptedAcknowledgements(uint16_t & outValue) const = 0;
+
+    /**
+     * @brief Retrieves the accepted terms and conditions acknowledgements version.
+     *
+     * @param[out] outValue The outValue of the accepted acknowledgements version.
+     * @return CHIP_ERROR On success returns CHIP_NO_ERROR, otherwise returns an error code.
+     */
+    virtual CHIP_ERROR GetTermsAndConditionsAcceptedAcknowledgementsVersion(uint16_t & outValue) const = 0;
+
+    /**
+     * @brief Sets the acceptance status of the terms and conditions.
+     *
+     * @param[in] inTCAcknowledgements The acknowledgements to accept.
+     * @param[in] inTCAcknowledgementsoutValue The version of the acknowledgements to accept.
+     * @return CHIP_ERROR On success returns CHIP_NO_ERROR, otherwise returns an error code.
+     */
+    virtual CHIP_ERROR SetTermsAndConditionsAcceptance(uint16_t inTCAcknowledgementsValue,
+                                                       uint16_t inTCAcknowledgementsoutValue) = 0;
+
+    /**
+     * @brief Clears the acceptance status of the terms and conditions.
+     *
+     * @return CHIP_ERROR On success returns CHIP_NO_ERROR, otherwise returns an error code.
```

Comment:
This is not useful documentation for the return.  Of course it returns CHIP_NO_ERROR on success and an error on error.

Just take out the @return lines if they're not documenting anything specific about error codes, please.